core/src/main/java/com/graphhopper/util/EdgeIteratorState.java

```diff
@@ -73,6 +74,16 @@ public void setBool(boolean reverse, IntsRef ref, boolean value) {
      */
     int getEdge();
 
+    /**
+     * Returns the edge key of the current edge. The edge id can be derived from the edge key by calling
+     * {@link GHUtility#getEdgeFromEdgeKey(int)}, but the edge key also contains information about the
+     * direction of the edge. The edge key is even when the edge is oriented in storage direction and odd
+     * otherwise. You can use the edge key to retrieve an edge state in the associated direction using
+     * {@link Graph#getEdgeIteratorStateForKey(int)}. Loop edges are always returned in 'forward' direction even when
+     * you use an odd edge key.
+     */
+    int getEdgeKey();
```

Comment:
In #2146 we call this `int getEdgeKey(boolean reverse)`. I would argue that the edge state has only one edge key and reversing the key is out of its scope, so I left out the reverse parameter. The reverse key can be obtained using `GHUtility.getReverseKey(edgeState.getEdgeKey())`, which of course is a bit less comfortable, especially in cases like this:

```java
int key = edgeState.getEdgeKey();
if (reverse)
    key = GHUtility.getReverseKey(key);
```

compared to:
```
int key = edgeState.getEdgeKey(reverse);
```

So not sure...

Reply:
Oh yeah, excellent. I‘m not a fan of the (boolean reverse) parameter anywhere, I just kept to the convention. Let‘s drop it.

Reply:
Ok so we keep `edgeState.getEdgeKey()` here :+1: